Framework/API/inc/MantidAPI/SpectrumDetectorMapping.h

```diff
@@ -53,7 +47,7 @@ class MANTID_API_DLL SpectrumDetectorMapping {
   using sdmap = std::unordered_map<specnum_t, std::set<detid_t>>;
 
 public:
-  explicit SpectrumDetectorMapping(const MatrixWorkspace *const workspace,
+  explicit SpectrumDetectorMapping(MatrixWorkspace_const_sptr &workspace,
```

Comment:
Is it safe to pass a `shared_ptr` as a reference to a public function like this?

Reply:
Good point! I'll try to find a solution.

Reply:
Maybe just drop the `&` and it should be OK.